java/fury-core/src/main/java/org/apache/fury/meta/ClassDefEncoder.java

```diff
@@ -146,36 +135,63 @@ static MemoryBuffer encodeClassDef(
       int currentClassHeader = (fields.size() << 1);
       if (classResolver.isRegistered(type)) {
         currentClassHeader |= 1;
-        buffer.writeVarUint32Small7(currentClassHeader);
-        buffer.writeVarUint32Small7(classResolver.getRegisteredClassId(type));
+        classDefBuf.writeVarUint32Small7(currentClassHeader);
+        classDefBuf.writeVarUint32Small7(classResolver.getRegisteredClassId(type));
       } else {
-        buffer.writeVarUint32Small7(currentClassHeader);
+        classDefBuf.writeVarUint32Small7(currentClassHeader);
         Class<?> currentType = getType(type, className);
         Tuple2<String, String> encoded = Encoders.encodePkgAndClass(currentType);
-        writePkgName(buffer, encoded.f0);
-        writeTypeName(buffer, encoded.f1);
+        writePkgName(classDefBuf, encoded.f0);
+        writeTypeName(classDefBuf, encoded.f1);
       }
-      writeFieldsInfo(buffer, fields);
+      writeFieldsInfo(classDefBuf, fields);
+    }
+    byte[] compressed =
+        classResolver
+            .getFury()
```

Comment:
I had an idea when I was looking at the code recently. 

Could `org.apache.fury.config.Config` be set to `org.apache.fury.resolver.SerializationContext`?
I think it is more reasonable to get `Config` from `SerializationContext` during runtime. It feels a bit weird to get it from `Fury` object.

This will reduce the burden on the `Fury` class.

Reply:
Good suggestion, I like the idea to get config from context. But other get context, user must get fury first, then get config from context. It will introduce an extra indirection

Reply:
I think it's not related to this PR, we can discuss it in another issue